boot/bootutil/src/encrypted.c

```diff
@@ -302,10 +306,19 @@ boot_enc_load(const struct image_header *hdr, const struct flash_area *fap,
     return rc;
 }
 
-int
+bool
 boot_enc_valid(const struct flash_area *fap)
 {
-    return enc_state[fap->fa_id - FLASH_AREA_IMAGE_PRIMARY].valid;
+    int rc;
+
+    rc = flash_area_id_to_image_slot(fap->fa_id);
+    if (rc < 0) {
+        /* can't get proper slot number - skip encryption, */
+        /* postpone the erro for a upper layer */
+        return false;
+    }
+
+    return enc_state[rc].valid;
```

Comment:
Since you are at it, maybe you should also make this field `bool`? :-P

Reply:
it is uint8 - less RAM. I would keep as is.